editor.planx.uk/src/routes/views/flowEditor.tsx

```diff
@@ -144,3 +163,45 @@ export const flowEditorView = async (req: NaviRequest) => {
     </FlowEditorLayout>
   );
 };
+
+const includedServices = [
+  getFOIYNPP(environment),
+  getRAB(environment),
+  getSubmission(environment),
+  getDiscretionary(environment),
+];
+
+export const generateAnalyticsLink = ({
+  environment,
+  flowId,
+  flowSlug,
+}: {
+  environment: Environment;
+  flowId: string;
+  flowSlug: string;
+}): string | undefined => {
+  let dashboardId: string | undefined;
+
+  for (const service of includedServices) {
```

Comment:
This is a custom implementation of `Array.prototype.find()` - it's probably easier / cleaner to reach for this built-in function.

```ts
const service = includedServices.find(service => 
  service.slugs.some(slug => flowSlug.includes(slug))
);
  
if (!service) return undefined;
```

Alternatively, with a callback - 

```ts
const hasMatchingSlug = (service) => 
  service.slugs.some(slug => flowSlug.includes(slug));
  
const service = includedServices.find(hasMatchingSlug);

if (!service) return undefined;
```